packages/react/src/components/InAppMessaging/InAppMessageDisplay/handleMessageLinkAction.ts

```diff
@@ -0,0 +1,29 @@
+import { ConsoleLogger as Logger } from '@aws-amplify/core';
+import { HandleMessageLinkAction } from '@aws-amplify/ui-react-core';
+
+const logger = new Logger('Notifications.InAppMessaging');
+
+const handleMessageLinkAction: HandleMessageLinkAction = (input) => {
+  let url: URL;
+
+  try {
+    url = new URL(input);
+  } catch {
+    logger.warn(`Unsupported url provided: ${input}`);
+    return;
+  }
+
+  const { protocol } = url;
+
+  const isHttpProtocol = protocol === 'http:';
```

Comment:
Non-blocking question: Curious why we even want to support the insecure HTTP protocol. With [HTTPS-only mode in FF](https://support.mozilla.org/en-US/kb/https-only-prefs), http sites will actually be blocked.

Reply:
Users can define either protocol in the message payload, so even if FF won't handle them we have to allow them